business-registry-dashboard/app/pages/index.vue

```diff
@@ -1,7 +1,78 @@
 <script setup lang="ts">
+const { t } = useI18n()
+const affStore = useAffiliationsStore()
+const brdModal = useBrdModals()
+const toast = useToast()
+
 definePageMeta({
   order: 0
 })
+
+// Token parsing
+const parseToken = (encodedToken: string): AffiliationToken => {
+  try {
+    const tokenObject = encodedToken.split('.')[0]
+    const decoded = atob(tokenObject as string)
+    return JSON.parse(decoded)
+  } catch (error) {
+    console.error('Failed to parse token:', error)
+    throw new Error('Invalid token format')
+  }
+}
+
+onMounted(async () => {
+  // Check for affiliation token from session storage
+  if (sessionStorage.getItem('affiliationToken')) {
+    const affiliationToken = sessionStorage.getItem('affiliationToken') as string
+    // Remove the token from session storage after it's used
+    sessionStorage.removeItem('affiliationToken')
+    try {
+      const parsedToken = parseToken(affiliationToken)
+      // Parse the URL and try to add the affiliation
+      parseUrlAndAddAffiliation(parsedToken, affiliationToken)
+      // Load affiliations to update the table
+      await affStore.loadAffiliations()
+    } catch (e) {
+      console.error('Error accepting affiliation invitation:', e)
+    }
+  }
+})
```

Comment:
I don't love this design of "check URL parameters + set session storage + reload + check session storage + handle token". It feels complicated and I keep thinking there are edge cases.

Can you check this design with someone else, like Travis?

Reply:
OK, will check with Travis now and see what he thinks.

Reply:
This shouldn't be an issue now

Reply:
Please resolve this conversation.